pkg/kv/kvserver/replica_init.go

```diff
@@ -229,19 +291,6 @@ func (r *Replica) loadRaftMuLockedReplicaMuLocked(desc *roachpb.RangeDescriptor)
 	}
 
 	r.setDescLockedRaftMuLocked(ctx, desc)
-
-	// Only do this if there was a previous lease. This shouldn't be important
-	// to do but consider that the first lease which is obtained is back-dated
-	// to a zero start timestamp (and this de-flakes some tests). If we set the
-	// min proposed TS here, this lease could not be renewed (by the semantics
-	// of minLeaseProposedTS); and since minLeaseProposedTS is copied on splits,
-	// this problem would multiply to a number of replicas at cluster bootstrap.
-	// Instead, we make the first lease special (which is OK) and the problem
-	// disappears.
-	if r.mu.state.Lease.Sequence > 0 {
-		r.mu.minLeaseProposedTS = r.Clock().NowAsClockTimestamp()
-	}
-
 	r.assertStateRaftMuLockedReplicaMuRLocked(ctx, r.store.Engine())
```

Comment:
This line can probably go away. I didn't like the invariant break conditional to `r.store.TestingKnobs().DisableEagerReplicaRemoval`, but I think it should not be the case on replica creation?

Reply:
Also, `assertStateRaftMuLockedReplicaMuRLocked` can be implemented using `loadReplicaState` now, but I'm not sure what to do about this invariant exception.